---
title: "chile-slides-201612-ppt-version"
author: "Ann Von Holle"
date: "December 9, 2016"
output: html_document
bibliography: lit.bib
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

system("java -version")
library('ReporteRs') # Load

```

```{r}
# see http://www.sthda.com/english/wiki/create-and-format-powerpoint-documents-from-r-software for more info

# Pick up any preferred ppt template
# Download a PowerPoint template file from STHDA website
#download.file(url="http://www.sthda.com/sthda/RDoc/example-files/r-reporters-powerpoint-template.pptx", destfile="r-reporters-powerpoint-template.pptx", quiet=TRUE)
options('ReporteRs-fontsize'= 18, 'ReporteRs-default-font'='Arial')
doc <- pptx(template="Presentation1.pptx" )

# info consistent across slides
foot.txt = "Kari North"

# define parProperties with list properties
unordered.list.level1 = parProperties(list.style = "unordered", 
	level = 1 )
unordered.list.level2 = parProperties(list.style = "unordered", 
	level = 2 )
unordered.list.level3 = parProperties(list.style = "unordered", 
	level = 3 )
options( "ReporteRs-fontsize" = 20 )

# Create a PowerPoint document
#doc = pptx( )

# Slide 1 : Title slide
#+++++++++++++++++++++++
doc <- addSlide(doc, "Title Slide")
doc <- addTitle(doc, "Lipid-related Genetic Variants And Lipid Outcomes In A Cohort Of Chilean Children")
#doc <- addSubtitle(doc, "R and ReporteRs package")
#doc <- addDate(doc)
doc <- addFooter(doc, foot.txt)
#doc <- addPageNumber(doc, "1/4")


# Slide 2 : Intro (1)
#+++++++++++++++++++++++
doc <- addSlide(doc, "Title and Content")
doc <- addTitle(doc,"Introduction: Dyslipidemia in children")
doc <- addParagraph(doc, 
                    value = "Dyslipidemia is an important risk factor for chronic cardiometabolic diseases.")
doc <- addParagraph(doc,
                    value = "Dsylipidemia represents any abnormal levels of lipid, lipoprotein, or apolipoprotein factors [@kwiterovich_clinical_2008], including:",
                    par.properties = unordered.list.level1,
                    append = T)
doc <- addParagraph(doc, 
                    value= c("total cholesterol (TC)", "low density lipoprotein cholesterol (LDL-C)", "triglycerides (TG)", 'high density lipoprotein choleseterol (HDL-C)'),
                    par.properties = unordered.list.level2,
                    append = T)

# Slide 3 : Intro (2)
#+++++++++++++++++++++++
doc <- addSlide(doc, "Title and Content")
doc <- addTitle(doc,"Introduction: Heritability of dyslipidemia")
doc <- addParagraph(doc, "Lipid traits are heritable (h$^2$ can exceed 0.50) [@goode_heritability_2007] and strongly associated with cardiovascular disease sustaining strong interest in genetic research.")
doc <- addParagraph(doc, "By 2013, researchers have identified more than 150 established loci influencing lipid levels in adults [@willer_discovery_2013].",
                    append=T)

# Slide 4: Intro (3)
# ++++++++++++++++++++++++++++++++++++
doc <- addSlide(doc, "Title and Content")
doc <- addTitle(doc,"Introduction: Lipid Variants")
addImage(doc, "fig1-willer-2013.png", width = 5,
         height = 5)
doc <- addFooter(doc, 
                    value = "Source: Willer, C. J., Schmidt, E. M., Sengupta, S., Peloso, G. M., Gustafsson, S., Kanoni, S., Abecasis, G. R. (2013). Discovery and refinement of loci associated with lipid levels. Nature Genetics, 45(11), 1274–1283. https://doi.org/10.1038/ng.2797.")

# Slide 5: Intro (4)
# ++++++++++++++++++++++++++++++++++
doc <- addSlide(doc, "Title and Content")
doc <- addTitle(doc,"Introduction: Generalizability across ancestral groups")
doc <- addParagraph(doc, value='Results found for adults have been found to generalize to children in a Finnish population [@tikkanen_association_2011; @buscot_combined_2016].')

doc = addParagraph(doc, value="Certain variants that associate with a trait such as dyslipdemia in one ancestral group may not associate in other groups [@carlson_generalization_2013].",
                   par.properties = unordered.list.level1,
                   append=T)

doc = addParagraph(doc, value = "Why? Differences across ancestral groups include:",
                   par.properties = unordered.list.level2,
                   append = T)
doc <- addParagraph(doc, 
                    value= c("Variant frequencies",
                             "Linkage disequilibrium"),
                    par.properties = unordered.list.level3,
                    append = T)

doc = addParagraph(doc,
                   value = "No research exists examining the association between genetic variants and dyslipidemia studies in Hispanic/Latino (HL) children",
                   append=T)


# Slide 5: Intro (5)
# ++++++++++++++++++++++++++++++++++
doc <- addSlide(doc, "Title and Content")
doc <- addTitle(doc,"Introduction: Primary Aim")
doc <- addParagraph(doc, value='Examine the association of known lipid variants with lipid traits identified in large study of adult participants from a Chilean infancy cohort of primarily European-descent.')

doc = addParagraph(doc, value="Compare results to those found in a Finnish population.",
                   append=T,
                   par.properties = unordered.list.level2)


# Slide 5: Intro (6)
# ++++++++++++++++++++++++++++++++++
doc <- addSlide(doc, "Title and Content")
doc <- addTitle(doc,"Methods: Association tests")

doc <- addParagraph(doc, 
                    value= "We assessed single variant associations using linear regression for high density lipoprotein cholesterol (HDL-C), low density lipoprotein cholesterol (LDL-C) and triglycerides (TG), assuming an additive genetic model, adjusted for sex and ancestry (via principal components).")

doc = addParagraph(doc,
                   value = "Sample Model for HDL",
                   append=T)

doc = addParagraph(doc,
                   value="HDL_i = b0 + SNP_i b1 + sex_i b2 + ANCESTRY_i B + e_i", # Note: won't take greek letters or latex values
                   append=T,
                   par.properties = unordered.list.level2)

doc = addParagraph(doc,
                   value= "SNP_i represents one SNP with “genotypes were coded as 0, 1, or 2 when directly genotyped or as a predicted allele dosage (range, 0–2) when imputed.” [@tikkanen_association_2011]",
                   append=T,
                   par.properties = unordered.list.level2)
    
doc = addParagraph(doc,
                   value= "Only six variants from the Chilean sample met the a priori threshold of power > 0.8 to detect an association based on effect sizes from GWAS [@teslovich_biological_2010].",
                   append=T)


# Write the document 
# ++++++++++++++++++++++++++++++++++++
writeDoc(doc, "chile-slides-201612-ppt-version.pptx")
```

